src/common/btree_class.c

```diff
@@ -248,15 +248,15 @@ nv_rec_alloc(struct btr_instance *tins, d_iov_t *key, d_iov_t *val,
 
 	rc = tins->ti_umm.umm_nospc_rc;
 
-	roff = umem_zalloc(&tins->ti_umm, sizeof(*r) + name_len);
+	roff = umem_zalloc(&tins->ti_umm, sizeof(*r) + name_len, 0);
```

Comment:
I understand all of these `0`s are meant to be replaced with an appropriate value calculated by the caller, aren't they?

I am a little bit worried one of these hardcoded `0` bucket identifiers might be omitted for the next steps. It might result in hard-to-track-down errors since everything will be right till bucket 0 will be able to serve these requests.

Please consider creating a `DUMMY_BACKET_ID 0` definition which will be meant for removal as soon as the users of these APIs are ready. This is to make sure no `0` will be left behind.

Reply:
I have added new umem macros to do the allocation from memory buckets and retained the old umem definition. I have also introduced default bucket id macro UMEM_DEFAULT_MBKT_ID.